### DISCUSSION TITLE
A paper looking custom CSS for sharing pages

### DISCUSSION THREAD

**Discussion by ysslang** [TriliumNext/Trilium#2681] [Show and tell]
## Screenshot
![image](https://user-images.githubusercontent.com/12163672/155253471-d804e6b3-1f17-4e8c-b6cc-5c9b1b17956a.png)

## Demo site
[My_own_trilium_site](https://trilium.ysslang.com:60443/share/share_css__paper)

# How to use
Reference to official sharing wiki https://github.com/zadam/trilium/wiki/Sharing#styling-the-shared-notes
1. Add the following code as a CSS code note. 
2. Share this note.
2. On the note you want to style, add a `~shareCss` relation to the CSS code note. It's recommended to add a `#shareOmitDefaultCss ` label . And make both labels inheritable.

## CSS code
Or subscribe this [gist](https://gist.github.com/ysslang/46e2a57ca95ba9c7368cbd255d1ac769#file-share_css__paper-css)
```
body {
    font-family: 'Lucida Grande', 'Lucida Sans Unicode', arial, sans-serif;
    background: #dfe6e9;
}

#layout {
    display: flex;
    margin: 0 auto;
}

#menu {
    width: 15%;
    margin: 60px 5px;
    overflow: auto;
}

#menu p {
    margin: 0;
}

#menu>p {
    font-weight: bold;
    font-size: 110%;
}

#menu ul {
    padding-left: 20px;
}


#main {
    width: 80%;
    max-width: 1200px;
    position: relative;
    background: #fff;
    margin: 60px auto;
    box-shadow: 0px 2px 38px rgba(0, 0, 0, 0.2);
    padding: 25px;
    overflow: auto;
}


#parentLink {
    float: right;
    margin-top: 20px;
}

#title {
    margin: 0;
    padding-top: 10px;
}

img {
    max-width: 100%;
}

pre {
    white-space: pre-wrap;
    word-wrap: anywhere;
}

iframe.pdf-view {
    width: 100%;
    height: 800px;
}

#toggleMenuButton {
    display: none;
    position: fixed;
    top: 8px;
    left: 5px;
    width: 1.4em;
    border-radius: 5px;
    border: 1px solid #aaa;
    font-size: 2rem;
    z-index: 10;
    height: auto;
    color: black;
    cursor: pointer;
}

#childLinks.grid ul {
    list-style-type: none;
    display: flex;
    flex-wrap: wrap;
    padding: 0;
}

#childLinks.grid ul li {
    width: 180px;
    height: 140px;
    padding: 10px;
}

#childLinks.grid ul li a {
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 100%;
    border: 1px solid #ddd;
    border-radius: 5px;
    justify-content: center;
    align-content: center;
    text-align: center;
    font-size: large;
}

#childLinks.grid ul li a:hover {
    background: #eee;
}

#childLinks.list ul {
    list-style-type: none;
    display: inline-flex;
    flex-wrap: wrap;
    padding: 0;
    margin-top: 5px;
}

#childLinks.list ul li {
    margin-right: 20px;
}

#noteClippedFrom {
    padding: 10px 0 10px 0;
    margin: 20px 0 20px 0;
    color: #666;
    border: 1px solid #ddd;
    border-left: 0;
    border-right: 0;
}

#toggleMenuButton::after {
    position: relative;
    top: -2px;
    left: 1px;
}

@media (max-width: 48em) {
    #layout.showMenu #menu {
        display: block;
        margin-top: 40px;
    }

    #toggleMenuButton {
        display: block;
    }

    #layout.showMenu #main {
        display: none;
    }

    #title {
        padding-left: 60px;
    }

    #layout.showMenu #toggleMenuButton::after {
        content: "«";
    }

    #toggleMenuButton::after {
        content: "»";
    }

    #menu {
        display: none;
    }
}
```

**Comment by sigaloid** [comment]
Very nice, I like it! What's the license on this CSS? I want to use it :p

**Reply by ysslang** [reply]
Well, it's based on the [official sharing CSS style code](https://github.com/zadam/trilium/blob/v0.50.2/src/public/stylesheets/share.css#L1) and this [CSS snippet](https://codepen.io/AlHakem/pen/BxXrKe). So it's free to use.

**Reply by fliptoback** [reply]
Hi there, could you please tell me how do I change the URL text link colour and also the other text colours on the sidebar? Thanks in advance.

**Reply by ysslang** [reply]
Simply use `a, a:link, a:visited, a:focus, a:hover, a:active { color: red }` to modify link text color. And CSS prefixed with a `nav#menu ` will chose only sidebar elements.

**Reply by fliptoback** [reply]
Thanks ysslang. Whilst I can find these sections:

1. html a { }
2. html a:visited { }

But I cant find these

a:link,
a:visited,
a:focus
a:hover
a:active

Where do I find these please?

Edit: I figured this out. Essentially added the html a: hover { } after the body { } section. Repeat the same for the other tag.

**Comment by Nriver** [comment]
Hi, does it have a name? I just added it into https://github.com/Nriver/awesome-trilium with the name `ysslang's theme` ;p

**Comment by CharLinHeng** [comment]
it's a good plugin
